src/Common/Thermodynamics/relations.jl

```diff
@@ -845,7 +845,7 @@ function supersaturation(
 ) where {FT <: Real}
 
     q_sat::FT = q_vap_saturation_generic(param_set, T, ρ; phase = Liquid())
-    q_vap::FT = q.tot - q.liq - q.ice
+    q_vap::FT = max(FT(0), q.tot - q.liq - q.ice)
```

Comment:
This comes up a number of times. We should probably make it a `Thermodynamics` function, which takes care of it being returned non-negative.